File: increasingring/src/main/java/com/baybaka/increasingring/receivers/IncomingCallReceiver.java
package com.baybaka.increasingring.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.baybaka.increasingring.Getter;
import com.baybaka.increasingring.RunTimeSettings;
import com.baybaka.increasingring.service.ServiceStarter;

import org.slf4j.LoggerFactory;

public class IncomingCallReceiver extends BroadcastReceiver {


    @Override
    public void onReceive(Context context, Intent intent) {

        RunTimeSettings timeChanges= ((Getter)context.getApplicationContext()).getRunTimeChanges();
        LoggerFactory.getLogger(IncomingCallReceiver.class.getSimpleName()).debug("IncomingCallReceiver onReceive. Listener registarion {}", timeChanges.isPhoneStateListenerRegistered());

        if (! timeChanges.isPhoneStateListenerRegistered()) {
            ServiceStarter.startServiceWithCondition(context);
        }

    }

}

File: increasingring/src/main/java/com/baybaka/increasingring/receivers/IncomingCallReceiver.kt
package com.baybaka.increasingring.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.baybaka.increasingring.di.Getter
import com.baybaka.increasingring.service.ServiceStarter
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class IncomingCallReceiver : BroadcastReceiver() {

    companion object{
        val LOG: Logger = LoggerFactory.getLogger(IncomingCallReceiver::class.java.simpleName)
    }


    override fun onReceive(context: Context, intent: Intent?) {

        val runTimeSettings = (context.applicationContext as Getter).runTimeSettings

        val registered = runTimeSettings.isPhoneStateListenerRegistered
        LOG.debug("IncomingCallReceiver onReceive. PhoneStateListener is registered = $registered")

        //service is killed - need to restart it
        if (!registered) {
            ServiceStarter.startServiceWithCondition(context)
        }

    }

}
